### PR TITLE
Install OpenSSL required for neuron's tqperf integration test

### PR DESCRIPTION
**Pull request by pramodk**
* Building mpi4py from source fails due to setuptools API changes
   see https://github.com/mpi4py/mpi4py/issues/157
 * OpenSSL is required to compile neuron from source with integration
   of tqperf. Wonder if tqperf should be disabled if openssl is not found.

### CODE REVIEW

**Comment by nrnhines** [comment]
@pramodk I think that if there is not openssl, then the tqperf test should fall back to only translating/compiling mod/invlfire.mod.  test1.py already has code to check for the existence of IntervalFireSHA. Note that ```ci-test.sh``` presently skips the latter if Darwin. I suppose one could test for success of ```$NRNHOME/bin/nrnivmodl -coreneuron -loadflags -lcrypto modx``` and if it failed, fall back to ```rm -r -f $CPU ; $NRNHOME/bin/nrnivmodl -coreneuron mod```

**Comment by pramodk** [comment]
> @pramodk I think that if there is not openssl, then the tqperf test should fall back to only translating/compiling mod/invlfire.mod. test1.py already has code to check for the existence of IntervalFireSHA. 

I was wondering about this - I was thinking that if OpenSSL doesn't exist then we can just skip tqperf test. Or, we should add two separate tests (at CMake level, just by changing some parameter) to distinguish between "non-crypto" and "crypto" test. This way we exactly know what has been run.

**Comment by nrnhines** [comment]
I guess it is a matter of judgment whether non-crypto success gives a sufficient increment of test validation to make it worth the effort. At least it covers the same code.  A greater degree of validation, though not the level of crypto, would be to have a simple checksum on input  and output event times. It would be nice to extend this to be a substantive gpu test. At present, the ARTIFICIAL_CELL test does not use the GPU.

**Comment by olupton** [comment]
Not sure we need the setuptools fix anymore, https://github.com/neuronsimulator/nrn-build-ci/runs/4688817356 ran against your NEURON branch and everything passed.

**Comment by pramodk** [comment]
Ok. Seems like there are already new setuptools releases to address the breaking change: https://setuptools.pypa.io/en/latest/history.html

> I guess it is a matter of judgment whether non-crypto success gives a sufficient increment of test validation to make it worth the effort.

@nrnhines: Just to clarify, CI of `nrn` repository does check tqperf with libcrypto etc. This `nrn-build-ci` is a standalone CI which checks building of NEURON with minimum/necessary packages. Hence, I think it's reasonable to skip tqperf test in this CI when libcrypto/OpenSSL is not found.

We can install OpenSSLhere but I think testing build with minimum required packages is helpful (and goal of this CI). Hence I created https://github.com/neuronsimulator/nrn/pull/1571.

File: .github/workflows/build-neuron.yml [hunk 1]
```diff
@@ -116,7 +116,7 @@ jobs:
         working-directory: ${{github.workspace}}
         run: |
           if [ -n "${{matrix.branch_or_tag}}" ]; then BRANCH_OPT="--branch=${{matrix.branch_or_tag}}"; fi
-          git clone --depth=1 --single-branch ${BRANCH_OPT} ${{github.server_url}}/${{github.repository_owner}}/nrn
+          git clone --depth=1 --single-branch -b prmaodk/tqperf-crypto ${{github.server_url}}/${{github.repository_owner}}/nrn
```
Comment (olupton) [inline .github/workflows/build-neuron.yml:119]:
```suggestion
          git clone --depth=1 --single-branch -b ${BRANCH_OPT} ${{github.server_url}}/${{github.repository_owner}}/nrn
```

File: .github/workflows/build-neuron.yml [hunk 2]
```diff
@@ -116,7 +116,7 @@ jobs:
         working-directory: ${{github.workspace}}
         run: |
           if [ -n "${{matrix.branch_or_tag}}" ]; then BRANCH_OPT="--branch=${{matrix.branch_or_tag}}"; fi
-          git clone --depth=1 --single-branch ${BRANCH_OPT} ${{github.server_url}}/${{github.repository_owner}}/nrn
+          git clone --depth=1 --single-branch -b ${BRANCH_OPT} ${{github.server_url}}/${{github.repository_owner}}/nrn
```
Comment (olupton) [inline .github/workflows/build-neuron.yml:119]:
```suggestion
          git clone --depth=1 --single-branch ${BRANCH_OPT} ${{github.server_url}}/${{github.repository_owner}}/nrn
```

**Review by olupton — APPROVED** [review]

File: scripts/install_redhat.sh [hunk 3]
```diff
@@ -4,4 +4,5 @@ CMD=$(command -v dnf || command -v yum)
 ${CMD} update -y
 ${CMD} install -y bison cmake diffutils dnf flex gcc gcc-c++ git \
   openmpi-devel libXcomposite-devel libXext-devel make readline-devel ncurses-devel \
-  ninja-build python3-devel python3-pip python3-wheel sudo which wget unzip
+  ninja-build python3-devel python3-pip python3-wheel sudo which wget unzip \
+  openssl-devel  # for tqperf integration test
```
Comment (olupton) [inline scripts/install_redhat.sh:8]:
```suggestion
  findutils openssl-devel  # for tqperf integration test
```

Comment (olupton) [reply scripts/install_redhat.sh:8]: https://github.com/neuronsimulator/nrn-build-ci/runs/4713219510?check_suite_focus=true#step:7:84483
https://github.com/neuronsimulator/nrn-build-ci/runs/4713219244?check_suite_focus=true#step:7:84371

**Comment by olupton** [comment]
We might like to install OpenSSL on macOS too (https://github.com/neuronsimulator/nrn-build-ci/runs/4717133195?check_suite_focus=true#step:6:363)

**Pull request merged**